lib/chef/provider/windows_task.rb

```diff
@@ -243,6 +242,37 @@ def task_need_update?
         false
       end
 
+      # Comparing random_delay values using ISO8601::Duration object Ref: https://github.com/arnau/ISO8601/blob/master/lib/iso8601/duration.rb#L18-L23
+      # di = ISO8601::Duration.new(65707200)
+      # ds = ISO8601::Duration.new('P65707200S')
+      # dp = ISO8601::Duration.new('P2Y1MT2H')
+      # di == dp # => true
+      # di == ds # => true
+      def random_delay_updated?
+        if new_resource.random_delay.nil?
+          false
+        elsif current_resource.random_delay.nil? && new_resource.random_delay == "PT0S" # when user sets random_dealy to 0 sec
+          false
+        elsif current_resource.random_delay.nil? && !new_resource.random_delay.nil?
+          true
+        else
+          ISO8601::Duration.new(current_resource.random_delay) != ISO8601::Duration.new(new_resource.random_delay)
+        end
+      end
+
+      # Comparing execution_time_limit values using Ref: https://github.com/arnau/ISO8601/blob/master/lib/iso8601/duration.rb#L18-L23
+      def execution_time_limit_updated?
+        if new_resource.execution_time_limit.nil?
+          false
+        elsif current_resource.execution_time_limit.nil? && new_resource.execution_time_limit == "PT0S" # when user sets random_dealy to 0 sec
+          false
+        elsif current_resource.execution_time_limit.nil? && !new_resource.execution_time_limit.nil?
```

Comment:
You've already checked to see if new_resource.execution_time_limit.nil? above so you know at this point it's not nil.